woocommerce/woocommerce-gutenberg-products-block
# Shipping and collection method block package support

This improves compatibility with multiple shipping packages by implementing the following:

- Hides the shipping/collection interface if some packages cannot be collected.
- Updates the "Shipping Method X package verbiage to "Shipment X" 
- Ensures that when shipping options are displayed, an option is automatically selected. This ensures when you toggle between pickup and collection, the correct option is checked.

Will add screenshots and test instructions on friday.

Fixes #7164

### Screenshots

Collection packages:

![Screenshot 2022-10-14 at 12 52 11](https://user-images.githubusercontent.com/90977/195840427-82a070ad-04e2-4d92-946b-e588733cec82.png)

Shipping packages:

![Screenshot 2022-10-14 at 12 53 16](https://user-images.githubusercontent.com/90977/195840653-e1f9fbe3-209b-45fb-a81f-40642ff2af81.png)

### Testing

**To quickly get packages up and running I installed the 3rd party plugin "Multiple Packages for WooCommerce". In this I set things up to split by shipping package, then ensured 2 products were in the cart that are assigned different shipping packages.**

1. Toggle on collection
    a. check that the "selected" method changes to the first collection method in the totals area
    b. check that only collection rates are visible
2. Toggle on delivery
    a. check that the "selected" method changes to the first shipping method in the totals area
    b. check that only shipping methods are shown
3. In "Multiple Packages for WooCommerce" settings, turn off local pickup for one of the shipping classes:

![Screenshot 2022-10-14 at 12 55 29](https://user-images.githubusercontent.com/90977/195841032-3549a8b4-17a2-4575-88c1-ba99b930083d.png)

4. Now go to checkout. Confirm there is no toggle for collection/delivery, and all options are shown in the original place below shipping address:

![Screenshot 2022-10-14 at 12 55 57](https://user-images.githubusercontent.com/90977/195841101-96de73bd-f4bb-4716-89cb-605921089359.png)
